indico/modules/categories/client/js/calendar.js

```diff
@@ -105,11 +107,22 @@ import CalendarLegend from './components/CalendarLegend';
       },
       events({start, end}, successCallback, failureCallback) {
         function updateCalendar(data) {
-          const attr = {category: 'categoryId', location: 'venueId', room: 'roomId'}[groupBy];
+          const attr = {
+            category: 'categoryId',
+            location: 'venueId',
+            room: 'roomId',
+            keywords: 'keywordId',
+          }[groupBy];
           if (!attr) {
             throw new Error(`Invalid "groupBy": ${groupBy}`);
           }
-          const filteredEvents = data.events.filter(e => !filteredLegendElements.has(e[attr] ?? 0));
+          const filteredEvents = data.events.filter(e => {
+            let result = !filteredLegendElements.has(e[attr] ?? 0);
+            if (filteringByKeyword() && e.keywords.length) {
+              result &&= !e.keywords.every(kw => filteredKeywords.has(kw));
```

Comment:
I honestly prefer the new modern approach. New things are meant to make our life easier, and what looks clearer today might get outdated with time, actually because they are replaced by better ways.